src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/SingleSpaceSeparatorCheck.java

```diff
@@ -234,61 +237,51 @@ private boolean isTextSeparatedCorrectlyFromPrevious(String line, int columnNo)
      *         not preceded by another space.
      */
-    private static boolean isSingleSpace(String line, int columnNo) {
-        return isSpace(line, columnNo) && !Character.isWhitespace(line.charAt(columnNo - 1));
+    private static boolean isSingleSpace(int[] line, int columnNo) {
+        return isSpace(line, columnNo) && !CommonUtil.isCodePointWhitespace(line, columnNo - 1);
     }
 
     /**
      * Checks if the {@code line} at {@code columnNo} is a space.
      *
-     * @param line The line in the file to examine.
+     * @param line unicode code point array of line in the file to examine.
      * @param columnNo The column position in the {@code line} to examine.
      * @return {@code true} if the character at {@code columnNo} is a space.
      */
-    private static boolean isSpace(String line, int columnNo) {
-        return line.charAt(columnNo) == ' ';
-    }
-
-    /**
-     * Checks if the {@code line} at {@code columnNo} is a whitespace character.
-     *
-     * @param line The line in the file to examine.
-     * @param columnNo The column position in the {@code line} to examine.
-     * @return {@code true} if the character at {@code columnNo} is a
-     *         whitespace.
-     */
-    private static boolean isWhitespace(String line, int columnNo) {
-        return Character.isWhitespace(line.charAt(columnNo));
+    private static boolean isSpace(int[] line, int columnNo) {
+        return line[columnNo] == ' ';
     }
 
     /**
      * Checks if the {@code line} up to and including {@code columnNo} is all
      * non-whitespace text encountered.
      *
-     * @param line The line in the file to examine.
+     * @param line unicode code point array of line in the file to examine.
      * @param columnNo The column position in the {@code line} to examine.
      * @return {@code true} if the column position is the first non-whitespace
      *         text on the {@code line}.
      */
-    private static boolean isFirstInLine(String line, int columnNo) {
-        return CommonUtil.isBlank(line.substring(0, columnNo));
+    private static boolean isFirstInLine(int[] line, int columnNo) {
+        return CodePointUtil.isBlank(Arrays.copyOfRange(line, 0, columnNo));
     }
 
     /**
      * Checks if the {@code line} at {@code columnNo} is the end of a comment,
      * '*&#47;'.
      *
-     * @param line The line in the file to examine.
+     * @param line unicode code point array of line in the file to examine.
      * @param columnNo The column position in the {@code line} to examine.
      * @return {@code true} if the previous text is a end comment block.
      */
-    private static boolean isBlockCommentEnd(String line, int columnNo) {
-        return line.substring(0, columnNo).trim().endsWith("*/");
+    private static boolean isBlockCommentEnd(int[] line, int columnNo) {
+        final int[] lineFromWhitespace = CodePointUtil.stripTrailing(
```

Comment:
Please wrap on `CodePointUtil`, if line length allows, it will make this easier to read. Also - can we think of a more descriptive name? Something like `strippedLine` or ...? Also, I am curious; isn't it good enough if we just check that the line contains `*/`?

Reply:
It will not work if we used `contains` as shown

```
src % cat Test.java
public class Test {

    private void method() { // ok
        String b = "das*/dasda";        if (!b.equals("sdsa")) {

        }
    }
}

src % java -jar checkstyle-10.1-SNAPSHOT-all.jar -c config.xml Test.java
Starting audit...
Audit done.
src % java -jar checkstyle-10.0-all.jar -c config.xml Test.java         
Starting audit...
[WARN] /Users/shabana/Documents/OpenSource/CheckStyleTest/src/src/Test.java:4:41: Use a single space to separate non-whitespace characters. [SingleSpaceSeparator]
Audit done.
```
In SNAPSHOT `contains` is used

Reply:
Resolved.